core/trino-main/src/test/java/io/trino/sql/query/TestRowFilter.java

```diff
@@ -111,6 +129,30 @@ public void init()
 
         assertions = new QueryAssertions(runner);
         accessControl = assertions.getQueryRunner().getAccessControl();
+
+        MockConnectorFactory mockMissingColumns = MockConnectorFactory.builder()
+                .withGetViews((s, prefix) -> ImmutableMap.<SchemaTableName, ConnectorViewDefinition>builder()
+                        .put(new SchemaTableName("default", "nation_view"), view)
+                        .build())
+                .withGetColumns(schemaTableName -> {
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_with_optional_column"))) {
+                        return TPCH_NATION_WITH_OPTIONAL_COLUMN;
+                    }
+                    throw new UnsupportedOperationException();
+                })
+                .withData(schemaTableName -> {
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_with_optional_column"))) {
+                        return TPCH_NATION_DATA;
+                    }
+                    throw new UnsupportedOperationException();
+                })
+                .withallowMissingColumnsOnInsert(true)
+                .build();
+
+        runnerMissingColumns.createCatalog(MOCK_CATALOG_MISSING_COLUMNS, mockMissingColumns, ImmutableMap.of());
+
+        assertionsMissingColumns = new QueryAssertions(runnerMissingColumns);
```

Comment:
You don't need another `QueryAssertions`. Just use a session object in your test to change the default catalog or prefix a table name with `catalog.schema` (like you already did).